consensus/blake3pow/consensus.go

```diff
@@ -432,6 +433,26 @@ func (blake3pow *Blake3pow) Prepare(chain consensus.ChainHeaderReader, header *t
 func (blake3pow *Blake3pow) Finalize(chain consensus.ChainHeaderReader, header *types.Header, state *state.StateDB, txs []*types.Transaction, uncles []*types.Header) {
 	// Accumulate any block and uncle rewards and commit the final state root
 	accumulateRewards(chain.Config(), state, header, uncles)
+
+	if common.NodeLocation.Context() == common.ZONE_CTX && header.ParentHash() == chain.Config().GenesisHash {
```

Comment:
technically I would prefer this to happen _before_ processing block one, but we can create an issue to revisit this later.

What we wanted:
```
* process genesis block
* load genesis allocations
* process block 1
```

```
* process genesis block
* process block 1
* load genesis allocations
```

Reply:
https://github.com/dominant-strategies/go-quai/issues/859